apps/pipelines/models.py

```diff
@@ -30,15 +30,19 @@ def invoke(self, input: PipelineState, session: ExperimentSession | None = None)
         from apps.pipelines.graph import PipelineGraph
 
         runnable = PipelineGraph.build_runnable_from_json(self.data)
-
+        # Django doesn't auto-serialize objects for JSON fields, so we need to copy the input and save the ID of
+        # the session instead of the session object.
+        input_copy = input.copy()
+        input_copy["experiment_session"] = session.id
         pipeline_run = PipelineRun.objects.create(
-            pipeline=self, input=input, status=PipelineRunStatus.RUNNING, log={"entries": []}, session=session
+            pipeline=self, input=input_copy, status=PipelineRunStatus.RUNNING, log={"entries": []}, session=session
         )
 
         logging_callback = PipelineLoggingCallbackHandler(pipeline_run)
         logging_callback.logger.info("Starting pipeline run")
         try:
             output = runnable.invoke(input, config=RunnableConfig(callbacks=[logging_callback]))
+            output["experiment_session"] = session.id
```

Comment:
You should change the signature of this method to not default to `None` for `session`, or handle that case here.

Reply:
I made [this change](https://github.com/dimagi/open-chat-studio/pull/474/commits/6ad8e2b8a4350c5b54224617bd78eeebb15e7356), which will essentially do what this line does, but only when the state has the session in it